### ISSUE TITLE
git lfs issue even after adding the file

### ISSUE THREAD

**Issue by kamlekar** [git-lfs/git-lfs#2947]
Hi,

I am working on Unity and trying to push a car model to server. (which is around 138 MB)

In my project folder, First I installed 

    git lfs install 

and then 

    git lfs track "*.obj"

then

    git add .gitattributes
    git add Assets/r8/r8_gt_obj.obj

Then 

    git commit -m "add r8 car"
    git push

But still I see this issue in console:

>"C:\Program Files\Git\bin\git.exe" push --recurse-submodules=check --progress "origin" refs/heads/master:refs/heads/master
Git LFS: (0 of 0 files, 1 skipped) 0 B / 0 B, 122.54 MB skipped     Counting objects: 16, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (15/15), done.
Writing objects: 100% (16/16), 32.20 MiB | 569.00 KiB/s, done.
Total 16 (delta 8), reused 0 (delta 0)
remote: Resolving deltas: 100% (8/8), completed with 4 local objects. remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
remote: error: Trace: c626b934ee3d9566532fcc8672a1f247
remote: error: See http://git.io/iEPt8g for more information.
remote: error: File Assets/r8/r8_gt_obj.obj is 119.80 MB; this exceeds GitHub's file size limit of 100.00 MB
To https://github.com/kamlekar/AR-Example.git
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'https://github.com/kamlekar/AR-Example.git'
Done

>Press Enter or Esc to close console...

Please help.

**Comment by kamlekar** [comment]
I just realised, I did the below command once:

    git filter-branch -f --index-filter 'git rm --cached --ignore-unmatch Assets/r8/r8_gt_obj.obj'

Is this causing the issue? 

if yes, please help me to fix this.

**Comment by kamlekar** [comment]
I also tried

    git reset --hard original/refs/heads/master

Didn't work!!

**Comment by ttaylorr** [comment]
Hi @kamlekar, thanks for opening this issue and I'm sorry that you're having some trouble! Here's my take on what's going on. 

As you posted above, when you push refs to GitHub, the changes are being rejected because the contents of your push contains a file that is too large. Confusingly, `git lfs track` _does not_ apply retroactively to files that already exist in your repository and match that pattern. As such, you will need to explicitly migrate them into Git LFS with `git lfs migrate import --include="*.obj"`.

(In general, I think that this is a useful tip, so I have opened up https://github.com/git-lfs/git-lfs/pull/2948 to include this in the `README.md` in an attempt to avoid further confusion).

In response to:

> I just realised, I did the below command once:
> 
> git filter-branch -f --index-filter 'git rm --cached --ignore-unmatch Assets/r8/r8_gt_obj.obj'

I don't believe that this would cause any issues, though I am not sure that that command alone would remove a large file from your repository. In order to do so, you can take a look at [this](https://help.github.com/articles/removing-sensitive-data-from-a-repository/#using-filter-branch) guide, which contains specific instructions regarding how to remove a file from your repository's history.

##

In short, I think that you need to run `git lfs migrate import --include="/path/to/file"` in order to retroactively migrate existing large files into Git LFS in order to decrease the size of your Git push.

If this isn't the case, please don't hesitate to ping me here and I'd be more than happy to reopen this issue. Thanks!

**Comment by kamlekar** [comment]
Hi @ttaylorr 

Thanks for your detailed explanation. I tried  `git lfs migrate import --include="*.obj"`
But I am getting the following error:

>Venkateshwar@Venkat MINGW64 /c/xampp/htdocs/AR-Example (master)
$ git lfs migrate import --include="*.obj"
migrate: Fetching remote refs: ..., done
migrate: Sorting commits: ..., done
migrate: Rewriting commits: 100% (0/0), done
migrate: Updating refs: ..., done
Error running git [-c filter.lfs.smudge= -c filter.lfs.clean= -c filter.lfs.process= -c filter.lfs.required=false checkout --force]: 'error: unable to unlink old 'Library/shadercompiler-UnityShaderCompiler.exe0.log': Invalid argument' 'exit status 1'

**Comment by ttaylorr** [comment]
> Thanks for your detailed explanation. I tried `git lfs migrate import --include="*.obj"`
> But I am getting the following error:

I haven't seen that before. I think I have an idea of what's going on there, but to pin things down, I'll need you to run one more command. Can you rerun this command with `GIT_TRACE=1` in your environment and post the output here?

**Comment by kamlekar** [comment]
Hi @ttaylorr 

When I tried today, it worked!

>$ git lfs migrate import --include="*.obj"
migrate: Fetching remote refs: ..., done
migrate: Sorting commits: ..., done
migrate: Rewriting commits: 100% (0/0), done
migrate: Updating refs: ..., done
migrate: checkout: ..., done

Not sure what was the problem before.

Thanks.

**Comment by kamlekar** [comment]
It didn't work first because Unity was open. When I closed and tried again, it worked.

**Comment by marcinczenko** [comment]
Hmm...I am trying `git lfs migrate import --include="*.psd"` but I am getting error:

```
migrate: Fetching remote refs: ..., done                                                                                                                                                           
Error in git rev-list --stdin --reverse --topo-order --do-walk --: exit status 128 fatal: bad revision '^refs/remotes/origin/v0.1.1'                                                               

migrate: Sorting commits: ..., done
```

**Comment by marcinczenko** [comment]
Ok, so I managed to move things forward a bit with

```
git lfs migrate import --include="*.psd" --exclude-ref=v0.1.1
```

but I still getting `remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.`....

**Comment by deepesh-vasthimal-cko** [comment]
I exactly did the same thing , see my below steps. Note : my file size is 201.36, Is that an issue. 

git lfs migrate import --include="*.a"  
git add /Users/deepesh.vasthimal/Documents/sample.a
git commit -m "add large .a files"
git push 

final step fails with below message, This is annoying as i have tried this entire day and it always fails even on new workspace. Can anyone please help. 

Enumerating objects: 12, done.
Counting objects: 100% (12/12), done.
Delta compression using up to 12 threads
Compressing objects: 100% (8/8), done.
Writing objects: 100% (9/9), 76.96 MiB | 1.53 MiB/s, done.
Total 9 (delta 3), reused 0 (delta 0)
remote: Resolving deltas: 100% (3/3), completed with 2 local objects.
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
remote: error: Trace: ******862ffdf3f7afc
remote: error: See http://git.io/iEPt8g for more information.
remote: error: File /Users/deepesh.vasthimal/Documents/sample.a is 201.36 MB; this exceeds GitHub's file size limit of 100.00 MB

**Comment by bk2204** [comment]
Try passing `--everything` to your `git lfs migrate import` command.  If that doesn't work, please open a new issue, since this one is closed.

**Comment by deepesh-vasthimal-cko** [comment]
When i pass --everything my repo then has around 143 ahead and 142 behind and post which i cant do any pull or push as i get below errors. 

hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
deepesh.vasthimal sampleApp % git pull 
fatal: refusing to merge unrelated histories

**Comment by bk2204** [comment]
When you run `git lfs migrate import` without `--no-rewrite`, you're necessarily rewriting history, so you'll almost always need to force push.

**Comment by deepesh-vasthimal-cko** [comment]
It worked thank you @bk2204

**Comment by dedi0003** [comment]
Thank you, I solved my problem using migrate. So strange, my file is 300 but when I use migrate and push it it becomes 600MB? However I am so relieve it finally can be pushed to remote

**Comment by albertobonanni** [comment]
Worked for me too. Thanks everyone!

**Comment by SharkWithLasers** [comment]
@bk2204 thanks! that also worked for me!

Perhaps the wiki tutorial page should be updated to include the need for a force push?

https://github.com/git-lfs/git-lfs/wiki/Tutorial#using-git-lfs-migrate

**Comment by bk2204** [comment]
That's a great suggestion.  I've updated the wiki.

**Comment by PhillipMaire** [comment]
> Try passing `--everything` to your `git lfs migrate import` command. If that doesn't work, please open a new issue, since this one is closed.

this should come with a warning as it can be difficult to undo

**Comment by Ebrahim-Ramadan** [comment]
> GIT_TRACE=1

whats `GIT_TRACE=1` command is trying to do?
@ttaylorr

**Comment by Ebrahim-Ramadan** [comment]
for anyone beignner for git lfs, read my [article](https://ebrahim-ramadan.vercel.app/blogs/GitLFS)

**Comment by AndreyKulish** [comment]
Thanks a lot !!!!!!!
